GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -8,10 +8,101 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
+    var allEpisodesinSeasons = arrOfSeasonsData
+    var arrOfSeasons: [[GOTEpisode]] {
+        guard let searchString = searchString else {
+            return allEpisodesinSeasons
+        }
+        guard searchString != "" else {
+            return allEpisodesinSeasons
+        }
+        let searchResults = filterForSearchResult(arr: GOTEpisode.allEpisodes, string: searchString)
+        let separateSeasons = makeArrofSeasons(allEp: searchResults, seasons: 7)
+        return separateSeasons
+        
+    }
+    var searchString: String? = nil {
+        didSet {
+            self.gOTTableView.reloadData()
+        }
+    }
+    
+    //MARK: -- TableView Methods
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        switch section % 2 {
```

Comment:
Im not sure if this switch case is necessary? why not just return arrOfSeasons[section].count